lib/include/openamp/remoteproc.h

```diff
@@ -395,6 +395,10 @@ struct remoteproc {
  *        memory may not be off.
  * @shutdown: shutdown the remoteproc and release its resources.
  * @notify: notify the remote
+ * @get_mem: get remoteproc memory I/O region by either name, virtual address,
+ *           physical address or device address.
+ *           Note: could return either internal managed instance or the 'buf'
+ *           argument as needed.
```

Comment:
This is not enough documentation. 

It needs to specify return values, and error return values, as well as describing the values for each input parameter and what the default value should be if it is ignored. 

The circumstances under which the 'buf' argument could be set also need to be documented.

Reply:
> This is not enough documentation.
> 
> It needs to specify return values, and error return values, as well as describing the values for each input parameter and what the default value should be if it is ignored.
> 

But all other functions don't follow your suggestioin. It's better to create a new patch to fix all place instead.

> The circumstances under which the 'buf' argument could be set also need to be documented.

'buf' always pass to the callback, but the implementation could fill and return 'buf', or ignore it and return the internal managed instance.

Reply:
@edmooring does my reply address your concern?

Reply:
> @edmooring does my reply address your concern?

It does not. 

Newly-added APIs should have good documentation from the start. If the documentation for older APIs should be improved, that can be addressed separately.

I have thought about it more, and the implementation should always fill in the 'buf' argument. That way an application developer knows they can always trust its value.

Reply:
Done.